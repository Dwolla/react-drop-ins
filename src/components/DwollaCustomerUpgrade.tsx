import type { FC } from 'react';
import type { UpdateCustomerAttributes } from '../@types/component-attributes';

const DwollaCustomerUpgrade: FC<UpdateCustomerAttributes<any>> = (props) => {
  return <dwolla-customer-update {...props} />;
};

export default DwollaCustomerUpgrade;
