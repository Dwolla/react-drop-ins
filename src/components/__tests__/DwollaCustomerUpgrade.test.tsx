import TestRenderer from 'react-test-renderer';
import DwollaCustomerUpgrade from '../DwollaCustomerUpgrade';

test('DwollaCustomerUpdate', () => {
  const tree = TestRenderer.create(<DwollaCustomerUpgrade customerId="123" />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('with optional props', () => {
  const tree = TestRenderer.create(
    <DwollaCustomerUpgrade
      customerId="123"
      correlationId="123abc"
      email="janedoe@email.com"
      firstName="Jane"
      lastName="Doe"
      ipAddress="255.255.255.255"
      privacy="www.yourprivacy.com"
      terms="www.yourterms.com"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
