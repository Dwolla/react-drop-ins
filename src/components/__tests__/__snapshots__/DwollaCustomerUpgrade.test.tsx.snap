// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DwollaCustomerUpdate 1`] = `
<dwolla-customer-update
  customerId="123"
/>
`;

exports[`with optional props 1`] = `
<dwolla-customer-update
  correlationId="123abc"
  customerId="123"
  email="janedoe@email.com"
  firstName="Jane"
  ipAddress="255.255.255.255"
  lastName="Doe"
  privacy="www.yourprivacy.com"
  terms="www.yourterms.com"
/>
`;
